app/main/helpers/search_save_helpers.py

```diff
@@ -38,3 +44,27 @@ def __init__(self, search_api_url, frameworks_by_slug, include_markup=False):
             filters.values(),
             lots_by_slug
         )
+
+
+def get_saved_search_temporary_message_status(project, framework, following_framework):
+    if not project['lockedAt']:
+        if following_framework['status'] in ['coming', 'open', 'pending', 'standstill']:
```

Comment:
Showing this message when the framework is `coming` feels super premature - it could be up to 3 months away at that point? I'd even question if `open` is too soon.

Be aware that when we show the messages we need to have the dates fairly locked in, which isn't always the case this early on. We should probably drop the message when the following framework is just `coming`, and we should consider probably just showing the month+year rather than exact date if it's in `open`.

Thoughts?

(For example DOS3 is currently coming but we have _no_ dates for it set up in the database yet)

Reply:
+1

Agree with Sam that showing messages like this too early is not helpful for users.

Reply:
Ah yeah, that's a a good point. Thinking about it I'd be tempted to only start showing the message once we get to pending. That would give a reasonable amount of time between the messages appearing and the new framework coming in. I'll chat to Patrick too, see what he thinks.

Reply:
Have had a chat with Patrick, he thinks that we shouldn't show the messages before we enter 'pending', and from that point on we should only show it if the date has been set. We could have something like `if status in ['pending', 'standstill'] and framwork.FrameworkLiveAtUTC:`

Although I've just noticed the dates can't be `null`. Do we have a rule on what we set dummy dates too? Are they wildly in the past/future? Could set a check on if the date is after/before a certain time?

Reply:
Have chatted further with Patrick re: always having a date set. We'e decided to only start showing the messages at the beginning of standstill, which should give buyers 6-8 working days to see the initial messages.

Do people think this is long enough. Bear in mind where we currently are in standstill, that these messages won't apply for a DOS framework, and what is probably going to happen with G11.

Reply:
I think I've set the dates for DOS3 to unix epoch, but there's no real standard. They are set by the developer when the new framework is posted to the API, so they can be anything.

We might just need to document that the framework should be created with as-accurate-as-possible dates, and that these need to be checked as the framework progresses.

Reply:
@karlchillmaid @samuelhwilliams ^^^

Reply:
Should have said, that we'll also be adding something to the run book and the dm manual to document that this date needs to be set.

Reply:
I think I agreed with your initial decision that pending+standstill is probably good. That should be something like 5 weeks. 10 days is _probably_ too short, given how it can take some time to actually compare services and that the process is fairly labour intensive. 5 weeks is probably longer than ideal, but I'd probably err on that side that too little time, at which point the banner could come too late for users to finish their procurement.

@karlchillmaid might have some ideas on which way to fall and how long it takes to compare services - I can't recall a length of time off the top of my head now...

Reply:
The issue with pending is that we may not have the date from CCS by then. We didn't for this iteration. (Pending + conditional check on the date being set) would be perfect, but comparing the date to epoch feels like it might break.

Reply:
Yeah you're right. That's tricky.

Let's go for this then. We can always iterate in the future if we get evidence that it's too late 👍

Reply:
Actually I was being dumb. Standstill is definitely the right time given that we keep G9 services eligible for purchase for ~4 weeks after the new iteration goes live.

As you were!